rfcs/046-column-details-access-and-side-panels.md

```diff
@@ -0,0 +1,120 @@
+Click on any of the active columns and the panel will slide out from the right side. You can close it by clicking on the x at the top right corner. 
+
+## UI/UX-level Explanation
+
+### Narrower table metadata section and column metadata indicators
+
+![Narrower table metadata section](../assets/046/narrower-table-metadata-section.png)
+
+* Table metadata panel on left will be narrower, taking up 3 out of 12 columns instead of 5
+* All normal/top-level columns will not have a chevron to expand column details within the list, but the columns with description, stats, unique values, upstream columns and/or downstream columns will have a clickable column name, shown in interactive (Indigo 60) color.
+* The chevron will now have a new purpose: to expand/collapse nested columns hierarchy. See more details about the new chevron interaction, how we will visualize nested columns in this new layout, also utilizing the right side panel, in the [Nested Columns RFC](./047-nested-columns.md).
+* After each clickable column name, small color-coded metadata indicator icons will show users which types of info are available for the column, even before they click on the column to see the details.
+* If a column does not have any description, stats, unique values, upstream columns and/or downstream columns info to show, the name will not be clickable, and therefore shown in regular black text.
+
+
+### Column details in right side panel
+
+![Column details in right side panel](../assets/046/right-side-panel.png)
+
+* Clicking on a column name will highlight (Gray 10) the selected column and open up the right side panel, which properly displays all column details. This right side panel will also take up 3 out of 12 columns. (See below for layout grid view)
+* Users can copy the full column name, copy column link, edit description, see all other column metadata, and/or click x to close the side panel.
+
+
+
+![Layout grid view](../assets/046/layout-grid-view.png)
+
+
+## Reference-level Explanation (aka Technical Details)
+
+We’ll update the layout to reduce the left sidebar and add a collapsible right sidebar. We’ll probably tackle it with a simple <aside> tag and a simple animation.
+
+The size of the table showing the column info will be reduced when the right panel opens to ensure the column types and actions are not hidden behind it. When the side panel opens, the table area will
+* Column names, nested column hierarchy (e.g. indentation), description, column type and action button will stay in the table area
+* Usage and badges will be hidden from the table area, but displayed in the right side panel. 
+
+
+## Drawbacks
+
+We have held off on major layout changes due to implementation cost and disruption of existing UI, but the column list area is getting crowded as we add more column metadata, and is not flexible enough to add the [nested columns improvements](./047-nested-columns.md) we want to make.
+
+## Alternatives
```

Comment:
I think what is also worth consideration is merging concepts of Machine Learning Feature and Column treat them as the same first-class citizen (with ML Feature as column type the same way we have same view for table and table view). For me in many ways ML Feature is very similar (if not identical) to Column - If I had to figure out how would dedicated column view look like, it would be 99% in line with what ML Feature view provides.

It also opens up possibility to take this even further in the future with providing actual resource hierarchy, something we found to be a limitation in Amundsen (like more clearer database > table > column hierarchy in the app with dedicated views) but would be helpful.

I like the idea of quick access to column details but from the perspective of giving it more space in the UI. Side pane, however, is not very scalable in my opinion - for example: if we wanted to provide graphs for column quality (like distribution of values etc.) it won't be very easy to do with sections that narrow.

Reply:
Re: side panel size.
I think this could be worked around in the future. If we needed more horizontal space, we could move that panel to be "overlapping" the table content and make it so it could grow more.

Reply:
Thanks so much for the feedback- we have considered this but decided the panel would be the optimal direction for now because we don't have enough metadata to show at column level. Even the side panel might be sparse for most columns. 
Also, we think the side panel is a valuable addition to preview content without leaving the page, whether we have a dedicated page for columns or not. The advantage is that you can quickly peek at the metadata without clicking into next level of resource and compare multiple columns in the list. As you see in the Future Possibilities, this can be applied to view table metadata from search results page. 
What we could do is add a note about possibility to add a dedicated column page, when we feel the need to scale due to overflowing column details in side panel.